src/ui/public/highlight/highlight_html.js

```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash';
+import angular from 'angular';
+import highlightTags from './highlight_tags';
+import htmlTags from './html_tags';
+
+export default function getHighlightHtml(fieldValue, highlights) {
+  let highlightHtml = (typeof fieldValue === 'object')
+    ? angular.toJson(fieldValue)
```

Comment:
I can't remember exactly why this was put in here, but if I remember correctly, `fieldValue` could be an Angular object (with `$$`-prefixed values) and that's why we used `angular.toJson`.